include/hipSYCL/glue/generic/host/collective_execution_engine.hpp

```diff
@@ -14,20 +14,18 @@
 #include "hipSYCL/sycl/libkernel/backend.hpp"
 
 /**
- * Due to an issue with Boost.Intrusive (used by Boost.Fiber), on Windows,
- * which is triggered in device pass of Clang CUDA, we may only use this in host pass.
- * This should not be a problem, as this implementation is anyways just required during host pass.
+ * Allow disabling fibers; and don't try using them in device pass.
  */
-#if !defined(HIPSYCL_NO_FIBERS) && !defined(SYCL_DEVICE_ONLY)
-#define HIPSYCL_HAS_FIBERS
+#if !defined(ACPP_NO_FIBERS) && !defined(SYCL_DEVICE_ONLY)
+#define ACPP_USE_FIBERS
 #endif
 
-#ifdef HIPSYCL_HAS_FIBERS
+#ifdef ACPP_USE_FIBERS
```

Comment:
Perhaps we should consider removing this macro in the future. Not sure if after this PR there's still a use case for not having fibers available?

Reply:
> Not sure if after this PR there's still a use case for not having fibers available?

Some weirdo Windows or microcontroller (#1720) build?